examples/keystore-management/src/components/RLNMembershipRegistration.tsx

```diff
@@ -0,0 +1,321 @@
+"use client";
+
+import { useState } from 'react';
+import { useRLN } from '../contexts/RLNContext';
+import { useWallet } from '../contexts/WalletContext';
+import { DecryptedCredentials } from '@waku/rln';
+
+export default function RLNMembershipRegistration() {
+  const { registerMembership, isInitialized, isStarted, rateMinLimit, rateMaxLimit, error, initializeRLN } = useRLN();
+  const { isConnected, address, chainId } = useWallet();
+  
+  const [rateLimit, setRateLimit] = useState<number>(rateMinLimit);
+  const [isRegistering, setIsRegistering] = useState(false);
+  const [isInitializing, setIsInitializing] = useState(false);
+  const [registrationResult, setRegistrationResult] = useState<{
+    success?: boolean;
+    error?: string;
+    txHash?: string;
+    warning?: string;
+    credentials?: DecryptedCredentials;
+  }>({});
+
+  const isLineaSepolia = chainId === 59141;
+
+  const handleRateLimitChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = parseInt(e.target.value);
+    setRateLimit(isNaN(value) ? rateMinLimit : value);
+  };
+
+  const handleInitializeRLN = async () => {
+    setIsInitializing(true);
+    try {
+      await initializeRLN();
+    } catch (err) {
+      console.error("Error initializing RLN:", err);
+    } finally {
+      setIsInitializing(false);
+    }
+  };
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+    
+    if (!isConnected) {
+      setRegistrationResult({ success: false, error: 'Please connect your wallet first' });
+      return;
+    }
+    
+    if (!isInitialized || !isStarted) {
+      setRegistrationResult({ success: false, error: 'RLN is not initialized' });
+      return;
+    }
+    
+    if (!isLineaSepolia) {
+      setRegistrationResult({ success: false, error: 'Please switch to Linea Sepolia network' });
+      return;
+    }
+    
+    setIsRegistering(true);
+    setRegistrationResult({});
+    
+    try {
+      setRegistrationResult({ 
+        success: undefined, 
+        warning: 'Please check your wallet to sign the registration message.' 
+      });
+      
+      const result = await registerMembership(rateLimit);
```

Comment:
one thing we discussed is to export into `JSON` 
https://github.com/waku-org/lab.waku.org/blob/cdd74f9f56beeff3c5f2013fa3be9ec380138338/examples/experimental/rln-identity/src/rln.js#L55

Reply:
to be the next PR